app/src/main/java/com/eventyay/organizer/core/auth/AuthActivity.java

```diff
@@ -41,6 +44,26 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
                 .replace(R.id.fragment_container, new StartFragment())
                 .commit();
         }
+
+        handleIntent(getIntent());
+    }
+
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        handleIntent(intent);
+    }
+
+    private void handleIntent(Intent intent) {
+        String appLinkAction = intent.getAction();
+        Uri appLinkData = intent.getData();
+        if (Intent.ACTION_VIEW.equals(appLinkAction) && appLinkData != null){
+            String token = appLinkData.toString().substring(
+                appLinkData.toString().indexOf('=') + 1);
```

Comment:
Add a separate class to parse the link and add tests

Reply:
Done.